sycl/test/abi/sycl_symbols_windows.dump

```diff
@@ -280,9 +280,9 @@
 ??0SampledImageAccessorBaseHost@detail@_V1@sycl@@QEAA@$$QEAV0123@@Z
 ??0SampledImageAccessorBaseHost@detail@_V1@sycl@@QEAA@AEBV0123@@Z
 ??0SampledImageAccessorBaseHost@detail@_V1@sycl@@QEAA@V?$range@$02@23@PEAXHHV?$id@$02@23@W4image_channel_type@23@W4image_channel_order@23@Uimage_sampler@23@AEBVproperty_list@23@@Z
-??0SubmissionInfo@detail@_V1@sycl@@QEAA@XZ

```

Comment:
We can't remove symbols as that breaks ABI.

Reply:
The [ABI policy](https://intel.github.io/llvm/developer/ABIPolicyGuide.html#changing-abi) says this is okay for experimental extensions, which is the current status of sycl_ext_oneapi_graph

> Note: Features clearly marked as experimental are considered as an exception to this guideline.

Reply:
First, you need to re-generate this file independently of this PR, so that there is no unrelated movement (or to eliminate such movement manually). This line doesn't seem to be related to the graphs at all.

Second, I'll let @stdale-intel , @gmlueck , @jbrodman decide if breaking ABI for the graphs is acceptable or not - there were lots of contradicting comments about that lately...

Reply:
Would this break the ABI for every application that uses the graph extension, or only in certain limited cases?

Reply:
> First, you need to re-generate this file independently of this PR, so that there is no unrelated movement (or to eliminate such movement manually).

Does this mean that using `abicheck.py` with `--mode dump_symbols` is no longer a valid method for updating these files?

> Would this break the ABI for every application that uses the graph extension, or only in certain limited cases?

I'm not 100% certain it breaks compatibility, but if it does it would break it for every application I think, as the change is on the `command_graph` class ultimately.

Reply:
> I'm not 100% certain it breaks compatibility, but if it does it would break it for every application I think, as the change is on the `command_graph` class ultimately.

I believe this is highly undesirable because a lot of code is using the graph API now.  I suggest checking to see if it really does break ABI compatibility.  If so, can we introduce this change under `-fpreview-breaking-changes`?

Reply:
I can't comment on that until the PR (abi test in particular) is split into two changes. Right now the diff for it is unreviewable.

Reply:
I've pushed my proposed changes to this PR for easier review, so I can roll back or change as required. The proposed changes remove any newly added symbols for the ABI tests, so the changes to that file are removed from the PR entirely now.

Reply:
I think that would work. That said, it's still unknown if your previous version even had the ABI break, because of the unreadability of the diff in the abi test.

Reply:
> I think that would work. That said, it's still unknown if your previous version even had the ABI break, because of the unreadability of the diff in the abi test.

I think it is probably an ABI break even if it didn't show in the symbols dump (FWIW, linux never showed any changes, only on Windows did it change the symbols to include `owner_less_base` in the affected classes). Adding a base class I believe changes the memory layout of the class, which is probably not a compatible change unless I am mistaken.

Reply:
Empty bases optimization is mandatory by C++ spec, I think. Of course, MSVC requires special treatment via `__SYCL_EBO` when inheriting from several classes at once.

That said, I wouldn't be surprised if MSVC mangling scheme is weird too.

Reply:
@gmlueck Could you take another look at this new approach which guards the changes? Thanks!

Reply:
My only comment is that we shouldn't break ABI.  @aelovikov-intel, I think you raised the issue originally.  Can you review and give your opinion on whether the ABI will be broken?

Reply:
I strongly suspect that the first version was the right thing to do, with a minor tweak of splitting the changes in the abi test in two (or manually modifying it to avoid unrelated changes).

Reply:
I've taken another look at the new windows symbols these changes originally introduced, and I think I was misreading them slightly. There are only new symbols added for the base class constructors and `ext_oneapi_owner_before()`. So considering the empty base optimizations I think functionally we are only adding new functions to the existing classes, which should be ABI compatible.

I've reverted the guarding changes and manually updated the windows symbols dumps to only contain the newly added symbols from this PR.